game/src/core/level/utils/Coordinate.java

```diff
@@ -1,52 +1,29 @@
 package core.level.utils;
 
 import core.utils.Point;
+import core.utils.Vector2;
 
 /**
- * Coordinate in the dungeon, based on array index.
+ * A record representing a 2D coordinate with x and y Integer components.
  *
- * <p>No getter needed. All attributes are public.
+ * <p>Provides utility methods for coordinate operations such as converting to a point, adding a
+ * vector, calculating the vector to another coordinate, and computing the distance between two
+ * coordinates.
+ *
+ * @param x The x component of the coordinate.
+ * @param y The y component of the coordinate.
+ * @see Point
+ * @see Vector2
  */
-public class Coordinate {
-
-  /** The x-Coordinate. */
-  public int x;
-
-  /** The y-Coordinate. */
-  public int y;
+public record Coordinate(int x, int y) {
 
   /**
-   * Create a new Coordinate.
+   * Create a new coordinate from another coordinate.
    *
-   * @param x x-Coordinate
-   * @param y y-Coordinate
+   * @param other The other coordinate.
    */
-  public Coordinate(int x, int y) {
-    this.x = x;
-    this.y = y;
-  }
-
-  /**
-   * Copy a coordinate.
-   *
-   * @param copyFrom Coordinate to copy
-   */
-  public Coordinate(final Coordinate copyFrom) {
-    x = copyFrom.x;
-    y = copyFrom.y;
-  }
-
-  @Override
-  public boolean equals(final Object o) {
-    if (!(o instanceof Coordinate other)) {
-      return false;
-    }
-    return x == other.x && y == other.y;
-  }
-
-  @Override
-  public int hashCode() {
-    return x + y;
+  public Coordinate(Coordinate other) {
```

Comment:
Wäre evtl. ein `static public of(Coordinate other)` interessanter? Dann brauche ich als Kunde nur noch `Coordinate.of(old)` statt `new Coordinate(old)`? Ist aber vermutlich Geschmackssache :)